### ISSUE TITLE
LanguageTool doesn't check anything in texts longer than character limit

### ISSUE THREAD

**Issue by jgsw** [Zettlr/Zettlr#4098]
### Description

LanguageTool imposes a maximum character limit on length of requests via its server: 20,000 for the basic plan, and 40,000 for the premium plan.  If a text is longer than this, Zettlr doesn't check it at all via LanguageTool.

It would be more elegant to check the part of the text that _is_ less than the character limit (20,000 or 40,000 characters, depending on the plan). The "document too long" warning could remain for the user. 

### Reproducing

Always reproducible.

### Zettlr Version

Stable (most recent version)

### Specify version

3.0.0 Beta1

### Installation Method

From the Website or GitHub

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Windows 10

### Additional Information

A possible enhancement request related to this would be to send longer documents in chunks for LanguageTool processing. This would allow a longer document to be checked, while respecting the character limit for each individual request. 

(This might be difficult to implement, and LanguageTool's limits on the number of characters that can be checked in a minute might create limitations for longer documents, even if this could be done.)

**Comment by nathanlesage** [comment]
I am thinking about that, but this involves more than a chunked checking, but also caching etc. Also, this is not a bug, this is intended behavior.

**Comment by tscheims1** [comment]
@jgsw 

You colud just use a local installation of a language tool server and get rid of the text limit. (Its not hard to setup, its basicly only running a shell script - https://dev.languagetool.org/http-server

**Comment by nathanlesage** [comment]
Sooo, in one of the recent commits, I started caching the document states, i.e., this will reduce the load to the servers and also make the linting feel snappier. However, I thought about your other suggestion, and I will not be implementing any document clipping and piecemeal-checking to circumvent the 20.000/80.000 character limit.

Speaking very strictly, doing so would have legal implications. As per LT's usage agreement, they will just block your IP if there are too many requests, and the amount of logic on my side with zero added benefit is just not worth getting into any legally muddy waters.

Hence, I defer you to the (IMO relatively easy to implement) "good" solutions to this: (a) Just run your own local server as explained by @tscheims1, or (b) utilize Zettlr's project feature and split up longer documents into their chapters.

I just finished a medium-sized paper of about 15 pages, and the character count tells me 30.000, so it should be incredibly difficult to exceed even the free limit with this – perfectly fine – strategy.
